src/hw-monitor.cpp

```diff
@@ -226,10 +225,39 @@ namespace librealsense
     }
 
 
-    void hw_monitor::get_gvd(size_t sz, unsigned char* gvd, uint8_t gvd_cmd) const
+    void hw_monitor::get_gvd( size_t sz,
+                              unsigned char * gvd,
+                              uint8_t gvd_cmd,
+                              const std::set< int32_t > * retry_error_codes ) const
     {
         command command(gvd_cmd);
-        auto data = send(command);
+        hwmon_response p_response = hwmon_response::hwm_Unknown;
+        auto data = send( command, &p_response );
+        if( p_response != hwm_Success )
+        {
+            // If we get an error code that match to the error code defined as require retry,
+            // we will retry the command until it succeed or we reach a timeout
+            bool should_retry = retry_error_codes && ! retry_error_codes->empty()
+                             && retry_error_codes->find( p_response ) != retry_error_codes->end();
+            if( should_retry )
+            {
+                static constexpr size_t RETRIES = 50;
+                for( int i = 0; i < RETRIES; ++i )
```

Comment:
If you're gonna check the termination inside, seems extra to do it here. Just do `while( true )` and increment inside?